exchanges/okcoin/okcoin.go

```diff
@@ -12,613 +12,1656 @@ import (
+	if err != nil {
+		return nil, err
+	} else if len(resp) == 0 {
+		return nil, fmt.Errorf("%w for instrument %s", errNoOrderbookData, instrumentID)
+	}
+	return &resp[0], nil
 }
 
-// GetAccountWithdrawalHistory retrieves all recent withdrawal records.
-func (o *OKCoin) GetAccountWithdrawalHistory(ctx context.Context, currency string) ([]WithdrawalHistoryResponse, error) {
-	requestURL := getWithdrawalHistory
-	if currency != "" {
-		requestURL += "/" + currency
+// GetCandlesticks retrieve the candlestick charts. This endpoint can retrieve the latest 1,440 data entries. Charts are returned in groups based on the requested bar.
+func (o *Okcoin) GetCandlesticks(ctx context.Context, instrumentID string, interval kline.Interval, after, before time.Time, limit int64) ([]CandlestickData, error) {
+	if instrumentID == "" {
+		return nil, errMissingInstrumentID
 	}
-	var resp []WithdrawalHistoryResponse
-	return resp, o.SendHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, accountSubsection, requestURL, nil, &resp, true)
+	params := url.Values{}
+	params.Set("instId", instrumentID)
+	var err error
+	if interval != kline.Interval(0) {
+		var intervalString string
+		intervalString, err = intervalToString(interval, false)
+		if err != nil {
+			return nil, err
+		}
+		params.Set("bar", intervalString)
+	}
+	if !after.IsZero() {
+		params.Set("after", strconv.FormatInt(after.UnixMilli(), 10))
+	}
+	if !before.IsZero() {
+		params.Set("before", strconv.FormatInt(before.UnixMilli(), 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	var resp []candlestickItemResponse
+	err = o.SendHTTPRequest(ctx, exchange.RestSpot, getCandlesticksEPL, http.MethodGet, typeMarket, common.EncodeURLValues("candles", params), nil, &resp, false)
+	if err != nil {
+		return nil, err
+	}
+	return ExtractCandlesticks(resp)
 }
 
-// GetAccountBillDetails retrieves the bill details of the wallet. All the information will be paged and sorted in reverse chronological order,
-// which means the latest will be at the top. Please refer to the pagination section for additional records after the first page.
-// 3 months recent records will be returned at maximum
-func (o *OKCoin) GetAccountBillDetails(ctx context.Context, request *GetAccountBillDetailsRequest) ([]GetAccountBillDetailsResponse, error) {
-	encodedRequest, err := encodeRequest(request)
+// GetCandlestickHistory retrieve history candlestick charts from recent years.
+func (o *Okcoin) GetCandlestickHistory(ctx context.Context, instrumentID string, after, before time.Time, bar kline.Interval, limit int64) ([]CandlestickData, error) {
+	if instrumentID == "" {
+		return nil, errMissingInstrumentID
+	}
+	params := url.Values{}
+	params.Set("instId", instrumentID)
+	var err error
+	if bar != kline.Interval(0) {
+		var intervalString string
+		intervalString, err = intervalToString(bar, false)
```

Comment:
Please make the `bool` of `false` a param, because wrapper functions `GetHistoricCandles` and `GetHistoricCandlesExtended` need it to be `true` for UTC alignment